src/API/Site/Controllers/MerchantCenter/SetupCompleteController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\EmptySchemaPropertiesTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use WP_REST_Request as Request;
+use WP_REST_Response as Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class SetupCompleteController
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter
+ */
+class SetupCompleteController extends BaseController {
+
+	use EmptySchemaPropertiesTrait;
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 */
+	public function register_routes() {
+		$this->register_route(
+			'mc/setup/complete',
+			[
+				[
+					'methods'             => TransportMethods::CREATABLE,
+					'callback'            => $this->get_setup_complete_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get the callback function for marking setup complete.
+	 *
+	 * @return callable
+	 */
+	protected function get_setup_complete_callback(): callable {
+		return function( Request $request ) {
+			do_action( 'woocommerce_gla_mc_setup_completed' );
```

Comment:
This works fine, but it just seems to me it kind of hides the purpose of what this controller is doing. I think the action hook to trigger additional tasks is handy. But I don't see why we can't call `set_completed_timestamp` directly here if it's changed to a public function. I have the same opinion about `set_contact_information_setup`, being set directly and `MerchantSetupCompleted` not needing to be a Registerable. Just my opinion though, we don't need to necessarily rewrite all this code.